addons/knowledge/tests/test_knowledge_article_permissions.py

```diff
@@ -330,12 +337,43 @@ def test_set_member_permission(self):
                            {self.partner_portal: 'none'})
 
         # downgrade a permission, should desynchronize from parent
+        writable_child = self.article_write_contents_children[0].with_env(self.env)
+        writable_grand_child = self.article_write_contents_children[1].with_env(self.env)
+        self.assertEqual(
+            writable_child._get_article_member_permissions()[writable_child.id][self.partner_employee_manager.id]['permission'],
+            'write',
+            'Share Panel: if an article inherits a permission, its children should inherit that permission too')
+        self.assertEqual(
+            writable_grand_child._get_article_member_permissions()[writable_grand_child.id][self.partner_employee_manager.id]['permission'],
+            'write',
+            'Share Panel: if an article inherits a permission, its grand children should inherit that permission too')
```

Comment:
As you are testing the same thing, you could factorize a bit:

```
for child in [writable_child, writable_grand_child]:
    self.assertEqual(...)
```

Same thing here below.

Reply:
I see you changed with "article_write_contents_children".

Is it equivalent to your previous one though?
From what I see in the code, "article_write_contents_children" only contains a single record (maybe I did not look properly).

Reply:
It is equivalent, `article_write_contents_children` is just written in two parts to be able to add the first child as parent of the second:
```
       cls.article_write_contents_children += cls.env['knowledge.article'].create([
            {'name': 'Child of child of writable through inheritance',
             'parent_id': cls.article_write_contents_children[0].id,
            },
        ])
```